### DISCUSSION TITLE
Jetstream Max Consumers per Stream

### DISCUSSION THREAD

**Discussion by beatrausch** [nats-io/nats-server#4797] [Q&A] [answered]
Hi all,

is there a limit of consumers per stream? 
In our use case we probably will have around 50 million consumers per stream. 

Regards,
Daniel

**Comment by derekcollison** [comment, marked as answer]
There can be limits set yes. Consumers can range from heavy weight state machines with consensus (NRG). Those will not support what you are looking for.

For what you are looking for, I would suggest getting in touch with us here at Synadia to help design the system. It's possible, but you can not use a traditional consumer here, and would need to look into stream re-publish and gap logic with a mux'd KV for sequence offsets etc.
